src/termwidget.cpp

```diff
@@ -238,6 +247,29 @@ void TermWidgetImpl::bell() {
     }
 }
 
+bool TermWidget::eventFilter(QObject * /*obj*/, QEvent * ev)
+{
+    if (ev->type() == QEvent::MouseButtonPress)
+    {
+        QMouseEvent *mev = static_cast<QMouseEvent*>(ev);
+        if ( mev->button() == Qt::MiddleButton )
+        {
+            impl()->pasteSelection();
+            if(Properties::Instance()->swapMouseButtons2and3)
+            {
+                    const QPoint& point = static_cast<const QPoint&>(mev->pos());
```

Comment:
Is a cast necesary here? These two lines can also be combined as `impl()->customContextMenuCall(mev->pos());`